src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/meta/ConstantPool.java

```diff
@@ -170,6 +170,19 @@ interface BootstrapMethodInvocation {
         List<JavaConstant> getStaticArguments();
     }
 
+    /**
+     * Gets the constant pool index of a static argument of a bootstrap specifier. Used when the list
+     * of static arguments in the {@link BootstrapMethodInvocation} is an {@code int[]}. The list has
+     * two elements. The first one is the number of arguments and the second one is the {@code cpi}.
```

Comment:
Are you referring to `BootstrapMethodInvocation.getStaticArguments()` whose return type is `List<JavaConstant>`, not `int[]`?

> The list has two elements.

That's not consistent with [the code](https://github.com/openjdk/jdk/blob/bd477810b176696e0fd043f5594663ebcf9884cf/src/hotspot/share/jvmci/jvmciCompilerToVM.cpp#L830-L835).

Reply:
> Are you referring to BootstrapMethodInvocation.getStaticArguments() whose return type is List<JavaConstant>, not int[]?

It is indeed the list returned by `BootstrapMethodInvocation.getStaticArguments()`. In this case, we have in fact a List<PrimitiveConstant> which corresponds to the int[], but it is true it is confusing, I will rephrase it.

> That's not consistent with [the code](https://github.com/openjdk/jdk/blob/bd477810b176696e0fd043f5594663ebcf9884cf/src/hotspot/share/jvmci/jvmciCompilerToVM.cpp#L830-L835).

In this case, the arguments come from [here](https://github.com/openjdk/jdk/blob/bd477810b176696e0fd043f5594663ebcf9884cf/src/hotspot/share/interpreter/bootstrapInfo.cpp#L205-L208), which are always only two elements.

Reply:
> In this case

In what case? When introducing new API like this, you need to use precise language that spells out possible values, assumptions, typical usages etc. What you have written is a good start but needs more work.